packages/web-components/src/components/carousel/carousel.ts

```diff
@@ -111,6 +114,39 @@ class DDSCarousel extends LitElement {
     }
   }
 
+  /**
+   * Handles card focus throughout pages.
+   *
+   * @param event The event.
+   */
+  @HostListener('shadowRoot:focusin')
+  // @ts-ignore: The decorator refers to this method but TS thinks this method is not referred to
+  private _handleFocus = async ({ target, relatedTarget }: FocusEvent) => {
+    const currentContains = target !== this && this.contains(target as DDSCard);
+    const oldNotContains = target !== this && !this.contains(relatedTarget as DDSCard);
+    const currentCardIndex = Array.from(this.children).indexOf(target as HTMLElement);
+
+    // keep current page if tabbing back into the carousel after previously moving pages
+    if (currentContains && oldNotContains && currentCardIndex === 0) {
+      (this.children[this.start] as HTMLElement).focus();
+      return;
+    }
+
+    if (currentContains) {
+      // going forwards, change page depending on card index
+      if (currentCardIndex >= this.start + this.pageSize) {
+        const nextStart = currentCardIndex - (currentCardIndex % this.pageSize);
```

Comment:
```suggestion
        // The `currentIndex` floored by `pageSize`
        const nextStart = currentCardIndex - (currentCardIndex % this.pageSize);
```

Or you can do: `const nextStart = Math.floor(currentCardIndex / this.pageSize) * this.pageSize`